graylog2-server/src/main/java/org/graylog2/shared/bindings/providers/OkHttpClientProvider.java

```diff
@@ -66,7 +75,26 @@ public OkHttpClient get() {
 
         if (httpProxyUri != null) {
             final Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(httpProxyUri.getHost(), httpProxyUri.getPort()));
-            clientBuilder.proxy(proxy);
+            final ProxySelector proxySelector = new ProxySelector() {
+                @Override
+                public List<Proxy> select(URI uri) {
+                    switch (uri.getHost().toLowerCase(Locale.ENGLISH)) {
```

Comment:
Maybe use `InetAddress` here, e. g. https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#isAnyLocalAddress-- and/or https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#isLoopbackAddress--